code/obj/artifacts/artifact_machines/warper.dm

```diff
@@ -0,0 +1,46 @@
+/obj/machinery/artifact/warper
+	name = "artifact warper"
+	associated_datum = /datum/artifact/warper
+
+/datum/artifact/warper
+	associated_object = /obj/machinery/artifact/warper
+	type_name = "Warper"
+	rarity_weight = 365
+	validtypes = list("ancient","martian","eldritch","precursor")
+	validtriggers = list(/datum/artifact_trigger/force,/datum/artifact_trigger/radiation,/datum/artifact_trigger/carbon_touch,/datum/artifact_trigger/silicon_touch,/datum/artifact_trigger/heat)
+	fault_blacklist = list(ITEM_ONLY_FAULTS,TOUCH_ONLY_FAULTS)
+	activ_text = "suddenly starts warping space around it!"
+	deact_text = "deactivates, and lays silent."
+	react_xray = list(15,75,90,3,"ANOMALOUS")
+	var/max_teleports = 4
+	var/teleports = 0 //how many times has the artifact moved
+	var/grab_range = 2 //at what range can we "grab" living people
+	var/teleport_range = 8
+	var/wormholer = FALSE // wormholes instead of random offsets
+	post_setup()
+		..()
+		if (prob(30))
+			wormholer = TRUE
+		max_teleports = rand(1,6)
+		grab_range = rand(2,5)
+		if (!wormholer)
+			teleport_range = rand(8,12)
+
+	effect_process(var/obj/O)
+		if (..())
+			return
+		if (inrestrictedz(O)) // AZONE puzzle speedrun %Any
+			O.ArtifactDeactivated()
+			return
+		var/turf/T = get_turf(O)
+		if (teleports == max_teleports)
+			O.ArtifactDeactivated()
+			teleports = 0
+			return
+		else
+			var/loc = (wormholer ? pick(wormholeturfs) : get_offset_target_turf(T, rand(-teleport_range, teleport_range), rand(-teleport_range, teleport_range)) )
+			playsound(O.loc, "warp", 50)
+			for (var/mob/living/M in orange(grab_range,O))
+				M.set_loc(get_offset_target_turf(loc, rand(-grab_range, grab_range), rand(-grab_range, grab_range)))
```

Comment:
isn't this supposed to be teleport range bruv

Reply:
teleport range is usually too far so no, its intended